tests/mocha/field_colour_test.js

```diff
@@ -0,0 +1,179 @@
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+  });
+  suite('fromJson', function() {
+    test('Empty', function() {
+      var colourField = new Blockly.FieldColour.fromJson({});
+      assertValueDefault(colourField);
+    });
+    test('Null', function() {
+      var colourField = new Blockly.FieldColour.fromJson({ colour:null });
+      assertValueDefault(colourField);
+    });
+    test('Undefined', function() {
+      var colourField = new Blockly.FieldColour.fromJson({ colour:undefined });
+      assertValueDefault(colourField);
+    });
+    test('Non-Parsable String', function() {
+      var colourField = new Blockly.FieldColour.fromJson({ colour:'bad' });
+      assertValueDefault(colourField);
+    });
+    test('#FFFFFF', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#FFFFFF' });
+      assertValue(colourField, '#ffffff', '#fff');
+    });
+    test('#ffffff', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#ffffff' });
+      assertValue(colourField, '#ffffff', '#fff');
+    });
+    test('#FFFF00', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#FFFF00' });
+      assertValue(colourField, '#ffff00', '#ff0');
+    });
+    test('#ffff00', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#ffff00' });
+      assertValue(colourField, '#ffff00', '#ff0');
+    });
+    test('#BCBCBC', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#BCBCBC' });
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+    test('#bcbcbc', function() {
+      var colourField = Blockly.FieldColour.fromJson({ colour: '#bcbcbc' });
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+  });
+  suite('setValue', function() {
+    test('#ffffff -> Null', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
+      colourField.setValue(null);
+      assertValueDefault(colourField);
+    });
+    test('#ffffff -> Undefined', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
+      colourField.setValue(undefined);
+      assertValueDefault(colourField);
+    });
+    test('#ffffff -> Non-Parsable String', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
+      colourField.setValue('bad');
+      assertValueDefault(colourField);
+    });
+    test('#ffffff -> #000000', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
+      colourField.setValue('#000000');
+      assertValue(colourField, '#000000', '#000');
+    });
+    test('#ffffff -> #bcbcbc', function() {
+      var colourField = new Blockly.FieldColour('#ffffff');
+      colourField.setValue('#bcbcbc');
+      assertValue(colourField, '#bcbcbc', '#bcbcbc');
+    });
+  });
+  suite('Validators', function() {
+    var colourField;
+    setup(function() {
+      colourField = new Blockly.FieldColour('#ffffff');
```

Comment:
Use a colour string other than the default so you can distinguish default vs previous value.